conda/plan.py

```diff
@@ -395,14 +395,16 @@ def is_root_prefix(prefix):
     return abspath(prefix) == abspath(context.root_dir)
 
 
-def add_defaults_to_specs(r, linked, specs, update=False):
+def add_defaults_to_specs(r, linked, specs, update=False, prefix=None):
     # TODO: This should use the pinning mechanism. But don't change the API:
     # cas uses it.
     if r.explicit(specs):
         return
     log.debug('H0 specs=%r' % specs)
     names_linked = {r.package_name(d): d for d in linked if d in r.index}
     mspecs = list(map(MatchSpec, specs))
+    if is_private_env(prefix):
```

Comment:
I know you've merged this, and technically it's fine, but you could have moved this up three lines.